packages/notifi-react/lib/context/NotifiTargetContext.tsx

```diff
@@ -80,18 +89,54 @@ const formatTelegramForSubscription = (telegramId: string) => {
 export type UpdateTargetInputs = <T extends 'form' | 'toggle'>(
   target: T extends 'form'
     ? Extract<Target, 'email' | 'phoneNumber' | 'telegram'>
-    : Extract<Target, 'discord' | 'slack'>,
-  value: T extends 'form' ? TargetInputFromValue : boolean,
+    : Extract<Target, 'discord' | 'slack' | 'wallet'>,
+  value: T extends 'form' ? { value: string; error?: string } : boolean,
 ) => void;
 
+type FormTargetRenewArgs = {
+  target: Extract<Target, 'email' | 'telegram' | 'phoneNumber'>;
+  value: string;
+};
+
+type ToggleTargetRenewArgs = {
+  target: Extract<Target, 'discord' | 'slack' | 'wallet'>;
+  value: boolean;
+};
+
+type TargetRenewArgs = FormTargetRenewArgs | ToggleTargetRenewArgs;
+
+const isFormTargetRenewArgs = (
+  args: TargetRenewArgs,
+): args is FormTargetRenewArgs => {
+  return (
+    args.target === 'email' ||
+    args.target === 'telegram' ||
+    args.target === 'phoneNumber'
+  );
+};
+
+const isToggleTargetRenewArgs = (
+  args: TargetRenewArgs,
+): args is ToggleTargetRenewArgs => {
+  return (
+    args.target === 'slack' ||
+    args.target === 'wallet' ||
+    args.target === 'discord'
+  );
+};
+
 export type NotifiTargetContextType = {
   isLoading: boolean;
   error: Error | null;
   updateTargetInputs: UpdateTargetInputs;
-  renewTargetGroup: () => Promise<void>;
+  renewTargetGroup: (args?: {
```

Comment:
Shall we  use `singleTargetRenewArgs` instead of `args` here?
1. It is more descriptive
2. consistent with the impl section